### PR TITLE
HPCC-14970 Return service BindingType in WsTopology.TpServiceQuery

### PR DESCRIPTION
**Pull request by wangkx**
Also fix an old bug for ServiceType in WsTopology.TpServiceQuery.

Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/smc/SMCLib/TpWrapper.cpp [hunk 1]
```diff
@@ -457,9 +456,16 @@ void CTpWrapper::getTpEspServers(IArrayOf<IConstTpEspServer>& list)
                 StringBuffer xpath;
                 xpath.appendf("EspService[@name='%s']", service);
                 IPropertyTree* pServiceNode = root->queryPropTree(xpath.str());
-                const char* serviceType = pServiceNode ? pServiceNode->queryProp("Properties/@type") : "INVALID";
-                pTpBinding->setServiceType(eqEsp);
+                const char* serviceType = pServiceNode ? pServiceNode->queryProp("Properties/@type") : NULL;
+                if (serviceType && *serviceType)
+                    pTpBinding->setServiceType(serviceType);
 
+                if (pServiceNode && (clientVersion >= 1.23))
```
Comment (afishbeck) [inline esp/smc/SMCLib/TpWrapper.cpp]: You don't have to pass in or check the version for setting fields unless there is a lot of overhead in getting, transforming or setting the information.  The interface will handle the versioning for you.  Checking the version is only an optimization.

File: esp/smc/SMCLib/TpWrapper.hpp [hunk 2]
```diff
@@ -170,7 +170,7 @@ class TPWRAPPER_API CTpWrapper : public CInterface
     void getTpEclCCServers(IPropertyTree* environmentSoftware, IArrayOf<IConstTpEclServer>& ServiceList, const char* name = NULL);
     void getTpEclAgents(IArrayOf<IConstTpEclAgent>& list, const char* name = NULL);
     void getTpEclSchedulers(IArrayOf<IConstTpEclScheduler>& ServiceList, const char* name = NULL);
-    void getTpEspServers(IArrayOf<IConstTpEspServer>& list);
+    void getTpEspServers(double clientVersion, IArrayOf<IConstTpEspServer>& list);
```
Comment (afishbeck) [inline esp/smc/SMCLib/TpWrapper.hpp]: See comment about versioning.  You don't need to pass this.

**Comment by wangkx** [comment]
@afishbeck I made the changes based on your comments. Please review.

**Comment by afishbeck** [comment]
@wangkx looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**